### PR TITLE
[VectorLayoutAnalysis] Add debug printing

### PR DESCRIPTION
**Pull request by Groverkss**
Add useful debug printing for layout analysis.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp [hunk 1]
```diff
@@ -925,3 +925,33 @@ VectorLayoutInterface VectorLayoutAnalysis::getLayout(Value val) {
   }
   return layout->getLayout();
 }
+
+void VectorLayoutAnalysis::print(raw_ostream &os) {
```
Comment (harsh-nod) [inline compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp]: The majority of this function is concerned with annotating ops with layouts. Could you rename this function to reflect that? Something like void annotateLayouts() that is called prior to printing. Something like
```
void VectorLayoutAnalysis::annotateLayouts() {
...
}

void VectorLayoutAnalysis::dump() {
  annotateLayouts();
  root->dump();
  llvm::errs() << "\n";
}
```

File: compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp [hunk 2]
```diff
@@ -925,3 +925,33 @@ VectorLayoutInterface VectorLayoutAnalysis::getLayout(Value val) {
...
+      Attribute layout = getLayout<Attribute>(result);
+      if (!layout) {
+        continue;
+      }
+
+      op->setAttr("layout_result_" + std::to_string(index), layout);
+    }
+  });
+
+  root->dump();
```
Comment (harsh-nod) [inline compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp]: The raw_ostream os is not being used here? Do we need it all?

Comment (Groverkss) [reply compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp]: Fixed. Used root->print(os) instead.

File: compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp (hunk 1 above)
Comment (Groverkss) [reply compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp]: Done

**Review by harsh-nod — APPROVED** [review]
lgtm! thanks!

**Pull request merged**